### PR TITLE
#5174: Uplifitng microbenchmarks to run on BH

### PR DESCRIPTION
**Pull request by abhullar-tt**
### Ticket
#5174 

### Problem description
Need to uplift these microbenchmark tests to run on BH

### Checklist
- [x] [Post commit CI passes](https://github.com/tenstorrent/tt-metal/actions/runs/11862536614)
- [x] [Microbenchmarks](https://github.com/tenstorrent/tt-metal/actions/runs/11862548557)

### CODE REVIEW

File: tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp [hunk 1]
```diff
@@ -512,6 +515,192 @@ void get_dram_reader_core_coords_wormhole_b0(
     all_cores_ordered = adj_core_logical_realloc;
 }
 
+void get_dram_reader_core_coords_blackhole(
+    tt_metal::Device* device, CoreRangeSet& all_cores, std::vector<CoreCoord>& all_cores_ordered) {
+    // hardcoded for bh
+    uint32_t full_grid_size_x = 17;
+    uint32_t full_grid_size_y = 12;
```
Comment (tt-aho) [inline tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp:522]: Can these values be taken from `device->grid_size()` instead of hardcoded here?
Also seems like WH and BH functions for `get_dram_reader_core_coords_*` are basically identical apart from hardcoding these values. Would it be a good idea to commonize it to reduce duplication?

File: tests/scripts/run_tests.sh [hunk 2]
```diff
@@ -379,7 +379,7 @@ set_up_chdir() {
     # The user might have multiple entries in their PYTHONPATH so we should try to find the right one
     IFS=':' read -ra ENTRIES <<< "$PYTHONPATH"
     for ENTRY in "${ENTRIES[@]}"; do
-      if [[ $ENTRY == *"tt-metal" ]]; then
+      if [[ $ENTRY == *"third-metal" ]]; then
```
Comment (tt-aho) [inline tests/scripts/run_tests.sh:382]: Is this change intended? If so why is it needed?

File: tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp (hunk 1 above)
Comment (abhullar-tt) [reply tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp:522]: yes it should be commonized and be uplifted to understand harvested columns. just copied and pasted to run the test

File: tests/scripts/run_tests.sh (hunk 2 above)
Comment (tt-rkim) [reply tests/scripts/run_tests.sh:382]: Do you work on two side `metal`s???

**Review by tt-rkim — APPROVED** [review]

**Review by yugaoTT — APPROVED** [review]
please make sure the existing perf for whb0 is not changed (especially the dram read one). Thanks

File: tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp [hunk 3]
```diff
@@ -327,10 +330,11 @@ void get_dram_reader_core_coords_grayskull(
...
 }
 
-void get_dram_reader_core_coords_wormhole_b0(
+void get_dram_reader_core_coords(
     tt_metal::Device* device, CoreRangeSet& all_cores, std::vector<CoreCoord>& all_cores_ordered) {
-    // hardcoded for wh_b0
-    uint32_t full_grid_size_y = 12;
+
+    uint32_t full_grid_size_x = device->grid_size().x;
+    uint32_t full_grid_size_y = device->grid_size().y;
```
[the remark this answers is not in the document]
Comment (abhullar-tt) [reply tests/tt_metal/tt_metal/perf_microbenchmark/8_dram_adjacent_core_read/test_dram_read.cpp:337]: yes, it pulls the grid size from the soc descriptor yamls

**Review by tt-aho — APPROVED** [review]

**Review by ubcheema — APPROVED** [review]
